### PR TITLE
Problem: Missing NFT projection and APIs

### LINKED ISSUE #412: Problem: Missing NFT projection and APIs

### PR DESCRIPTION
**Pull request by calvinaco**
Solution: (Fix #412) Add NFT projection and APIs

### CODE REVIEW

File: migrations/20210514174506_view_nft_token_transfers.up.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE view_nft_token_transfers (
+    id BIGSERIAL,
```
Comment (cdc-Hitesh) [inline migrations/20210514174506_view_nft_token_transfers.up.sql:2]: Does it make sense to store the `view_nft_tokens` row ID here in this table?

File: projection/nft/nft.go [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+		fmt.Sprintf("-:-:-:%s", tokenRow.Owner),
+		fmt.Sprintf("%s:-:-:%s", tokenRow.DenomId, tokenRow.Owner),
+		fmt.Sprintf("-:%s:-:%s", tokenRow.Drop, tokenRow.Owner),
+		fmt.Sprintf("-:-:%s:%s", tokenRow.Minter, tokenRow.Owner),
+		fmt.Sprintf("%s:%s:-:%s", tokenRow.DenomId, tokenRow.Drop, tokenRow.Owner),
+		fmt.Sprintf("%s:-:%s:%s", tokenRow.DenomId, tokenRow.Minter, tokenRow.Owner),
+		fmt.Sprintf("-:%s:%s:%s", tokenRow.Drop, tokenRow.Minter, tokenRow.Owner),
+		fmt.Sprintf("%s:%s:%s:%s", tokenRow.DenomId, tokenRow.Drop, tokenRow.Minter, tokenRow.Owner),
+	}, 1); incrementErr != nil {
+		return fmt.Errorf("error decrementing NFT token total: %v", incrementErr)
```
Comment (cdc-Hitesh) [inline projection/nft/nft.go:289]:
```suggestion
		return fmt.Errorf("error incrementing NFT token total: %v", incrementErr)
```

File: projection/nft/nft.go [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+	committed = true
+	return nil
+}
+
+func (nft *NFT) updateToken(tokensView *view.Tokens, tokenRow view.TokenRow, tokensTotalView *view.TokensTotal) error {
+	if updateTokenErr := tokensView.Update(tokenRow); updateTokenErr != nil {
+		return fmt.Errorf("error updating NFT token row: %v", updateTokenErr)
+	}
+
+	if decrementErr := tokensTotalView.DecrementAll([]string{
```
Comment (cdc-Hitesh) [inline projection/nft/nft.go:267]: What's happening here, we are decrementing and then incrementing `tokensTotalView` by 1.
Sorry, Could you brief me?

File: config/config.mainnet.toml [hunk 4]
```diff
@@ -75,4 +75,6 @@ enables = [
     "Transaction",
     "Validator",
     "ValidatorStats",
+    "NFT",
+#    "CryptoComNFT",
```
Comment (cdc-Hitesh) [inline config/config.mainnet.toml:79]: CryptoComNFT will be enabled when?

**Review by cdc-Hitesh — COMMENTED** [review]
Fantastic work.
Left a few doubts and comments.

File: migrations/20210514174506_view_nft_token_transfers.up.sql (hunk 1 above)
Comment (calvinaco) [reply migrations/20210514174506_view_nft_token_transfers.up.sql:2]: This id is an auto-incrementing id of the `view_nft_token_transfers`, it has nothing to do with `view_nft_tokens`.

File: projection/nft/nft.go (hunk 3 above)
Comment (calvinaco) [reply projection/nft/nft.go:260]: Good catch, it is trying to update previous owner's entry then increment the new one. Code updated.

File: config/config.mainnet.toml (hunk 4 above)
Comment (calvinaco) [reply config/config.mainnet.toml:79]: "CryptoComNFT" is dedicated for Crypto.com based NFT. So by default it is disabled. If interested party are running Crypto.com compatible NFT formats they can disable "NFT" and enable it instead.

Comment (calvinaco) [reply config/config.mainnet.toml:79]: The logic's here
https://github.com/crypto-com/chain-indexing/pull/413/files#diff-36233cad5c04d2ed21a0945e0bb73cadb627342617e62ed07ca27184b8bf1d84R48

File: migrations/20210514174506_view_nft_token_transfers.up.sql (hunk 1 above)
Comment (cdc-Hitesh) [reply migrations/20210514174506_view_nft_token_transfers.up.sql:2]: No, I mean just something like a foreign key. Though it increases complexity.

**Pull request merged**